src/fidesops/service/connectors/saas_query_config.py

```diff
@@ -144,29 +180,77 @@ def map_param_values(
                 f"Unable to replace placeholders in body for the '{self.action}' request of the '{self.collection_name}' collection."
             )
 
+        # format the body based on the content type
+        updated_headers, formatted_body = self.format_body(headers, body)
+
         return SaaSRequestParams(
             method=current_request.method,
             path=path,
-            headers=headers,
+            headers=updated_headers,
             query_params=query_params,
-            json_body=json.loads(body) if body else update_values,
+            body=formatted_body,
         )
 
+    @staticmethod
+    def format_body(
+        headers: Dict[str, Any],
+        body: Optional[str],
+    ) -> Tuple[Dict[str, Any], Optional[str]]:
+        """
+        Builds the appropriately formatted body based on the content type,
+        adding application/json to the headers if a content type is not provided.
+        """
+
+        if body is None:
+            return headers, None
+
+        content_type = next(
+            (
+                value
+                for header, value in headers.items()
+                if header.lower() == "content-type"
+            ),
+            None,
+        )
+
+        # add Content-Type: application/json if a content type is not provided
+        if content_type is None:
+            content_type = "application/json"
+            headers["Content-Type"] = content_type
+
+        output: Optional[str] = None
```

Comment:
This variable is not being used, since you're overriding it in every instance below, or raising an error.